src/shared/specifier.c

```diff
@@ -219,19 +224,19 @@ static int specifier_os_release_common(const char *field, char **ret) {
 }
 
 int specifier_os_id(char specifier, const void *data, const void *userdata, char **ret) {
-        return specifier_os_release_common("ID", ret);
+        return specifier_os_release_common((const char *) data, "ID", ret);
```

Comment:
the cast should not be needed?

Reply:
I changed the argument for `specifier_os_release_common`, so its needed bc of that